js_modules/dagit/src/Materialization.tsx

```diff
@@ -1,88 +1,59 @@
 import * as React from "react";
 import styled from "styled-components";
-import { ROOT_SERVER_URI } from "./Util";
-import { Colors, Spinner } from "@blueprintjs/core";
+import { Toaster, Colors, Position, Intent } from "@blueprintjs/core";
 
-function fileLocationToHref(location: string) {
-  if (location.startsWith("/")) {
-    return `file://${location}`;
-  }
-  return location;
-}
+const SharedToaster = Toaster.create({ position: Position.TOP }, document.body);
 
 interface MaterializationProps {
   fileLocation: string;
   fileName: string;
 }
-interface MaterializationState {
-  opening: boolean;
-}
-
-export class Materialization extends React.Component<
-  MaterializationProps,
-  MaterializationState
-> {
-  state = {
-    opening: false
-  };
-
-  _openingTimeout?: NodeJS.Timeout;
-
-  componentWillUnmount() {
-    if (this._openingTimeout) {
-      clearTimeout(this._openingTimeout);
-    }
-  }
-
-  onOpenFileLink = async (event: React.MouseEvent<HTMLAnchorElement>) => {
-    const url = event.currentTarget.href;
-
-    // If the file's location is a path, tell dagit to open it on the machine.
-    if (url.startsWith("file://")) {
-      event.preventDefault();
 
-      // Display a loading spinner to give Dagit and the file handler (eg: jupyter notebooks)
-      // time to open the file. Otherwise the user might just think its not working. Just
-      // set to an arbitrary amount of time - jupyter is slow.
-      this.setState({ opening: true });
-      this._openingTimeout = setTimeout(() => {
-        this.setState({ opening: false });
-      }, 2000);
+function isFilePath(url: string) {
+  return url.startsWith("file://") || url.startsWith("/");
+}
+export class Materialization extends React.Component<MaterializationProps> {
+  onCopyPath = async (event: React.MouseEvent<HTMLAnchorElement>) => {
+    event.preventDefault();
 
-      const path = url.replace("file://", "");
-      const result = await fetch(
-        `${ROOT_SERVER_URI}/dagit/open?path=${encodeURIComponent(path)}`
-      );
+    const url = event.currentTarget.getAttribute("href") || "";
```

Comment:
This is worthy of it's own comment (and maybe isolate the DOM manipulations in their own function

Reply:
Hmm ok, I'll just get rid of this and bind the url to the function call. 👍